MapboxCoreNavigation/URLSession.swift

```diff
@@ -0,0 +1,59 @@
+import Foundation
+import MapboxDirections
+
+extension URLSession {
+    /**
+     :nodoc:
+     
+     The user agent string for any HTTP requests performed directly within MapboxCoreNavigation or MapboxNavigation.
+     */
+    public static let userAgent: String = {
```

Comment:
Is there any specific reason we use custom `User-Agent`? (`NSURLSession`sets `User-Agent` by default). I can also see that `Directions.swift` uses pretty similar approach when creating `userAgent` so might make sense to have some common way of creating it.

Reply:
In general, we should always customize the user agent string to provide more context in the server logs than the application and Core Foundation versions, which is what URLSession provides by default. This particular user agent string is specific to requests made by MapboxCoreNavigation or MapboxNavigation.